hazelcast/src/main/java/com/hazelcast/multimap/impl/MultiMapContainerSupport.java

```diff
@@ -49,15 +51,19 @@ public MultiMapValue getOrCreateMultiMapValue(Data dataKey) {
             return multiMapValue;
         }
         // create multiMapValue
-        final MultiMapConfig.ValueCollectionType valueCollectionType = config.getValueCollectionType();
-        final Collection<MultiMapRecord> collection = createCollection(valueCollectionType);
+        final Collection<MultiMapRecord> collection = getNewRecordsCollection();
         multiMapValue = new MultiMapValue(collection);
 
         multiMapValues.put(dataKey, multiMapValue);
 
         return multiMapValue;
     }
 
+    protected Collection<MultiMapRecord> getNewRecordsCollection() {
+        final MultiMapConfig.ValueCollectionType valueCollectionType = config.getValueCollectionType();
```

Comment:
I would not  declare the local variable. makes the method more complex than needs to be.